### PR TITLE
ref(slack): Refactor Link Identity View

### PR DESCRIPTION
**Pull request by iamrajjoshi**
https://www.notion.so/sentry/Slack-ff5b8ab4fd1e4760ab829438ce97ce15

Addresses `bot-3` task.

Here, I refactored the view to have two separate methods for get and post. I added bunch of logging and metrics & improved tests.

### CODE REVIEW

File: src/sentry/integrations/slack/views/link_identity.py [hunk 1]
```diff
@@ -40,33 +46,72 @@ class SlackLinkIdentityView(BaseView):
...
+        try:
+            Identity.objects.link_identity(
+                user=request.user, idp=idp, external_id=params["slack_id"]
             )
-
-        Identity.objects.link_identity(user=request.user, idp=idp, external_id=params["slack_id"])
+        except IntegrityError:
+            _logger.exception("slack.link.integrity_error")
+            metrics.incr(
+                self._METRICS_FAILURE_KEYKEY + ".post.identity.integrity_error",
```
Comment (ykamo001) [inline src/sentry/integrations/slack/views/link_identity.py:108]: might be a typo here

Comment (iamrajjoshi) [reply src/sentry/integrations/slack/views/link_identity.py:108]: fixed

File: src/sentry/integrations/slack/views/link_identity.py [hunk 2]
```diff
@@ -40,33 +46,72 @@ class SlackLinkIdentityView(BaseView):
...
+    _METRICS_FAILURE_KEY = "sentry.integrations.slack.link_identity_view.failure"
+
     @method_decorator(never_cache)
-    def handle(self, request: Request, signed_params: str) -> HttpResponse:
+    def dispatch(self, request: HttpRequest, *args, **kwargs) -> HttpResponseBase:
         try:
+            signed_params = kwargs.pop("signed_params")
             params = unsign(signed_params)
-        except (SignatureExpired, BadSignature):
+        except (SignatureExpired, BadSignature) as e:
```
Comment (ykamo001) [inline src/sentry/integrations/slack/views/link_identity.py:57]: what about KeyError here if the user does not pass in `signed_params`? Should propably return a 400

File: src/sentry/integrations/slack/views/link_identity.py [hunk 3]
```diff
@@ -40,33 +46,72 @@ class SlackLinkIdentityView(BaseView):
...
+            request=request,
+            context={"organization": organization, "provider": integration.get_provider()},
+        )
+
+    def post(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        params = kwargs["params"]
+        organization, integration, idp = (
+            params["organization"],
+            params["integration"],
+            params["idp"],
```
Comment (ykamo001) [inline src/sentry/integrations/slack/views/link_identity.py:98]: are we doing schema/contract checks anywhere and returning the proper error code?

Comment (ykamo001) [reply src/sentry/integrations/slack/views/link_identity.py:98]: Can we use a serializer or a dataclass and set this up to be contract based?

File: src/sentry/integrations/slack/views/link_identity.py [hunk 4]
```diff
@@ -40,47 +48,102 @@ class SlackLinkIdentityView(BaseView):
...
+            _logger.exception(
+                "get_identity_error", extra={"integration_id": params["integration_id"]}
+            )
+            metrics.incr(self._METRICS_FAILURE_KEY + ".get_identity", sample_rate=1.0)
+            raise
+
+        _logger.info("get_identity_success", extra={"integration_id": params["integration_id"]})
+        metrics.incr(self._METRICS_SUCCESS_KEY + ".get_identity", sample_rate=1.0)
+        params.update({"organization": organization, "integration": integration, "idp": idp})
+        return super().dispatch(request, params=params)
```
Comment (ykamo001) [inline src/sentry/integrations/slack/views/link_identity.py:91]: do we need to update params or could we just pass them into super().dispatch() as specific key word arguments? `super().dispatch(org=organization)`

File: src/sentry/integrations/slack/views/link_identity.py [hunk 5]
```diff
@@ -40,47 +48,102 @@ class SlackLinkIdentityView(BaseView):
...
-                context={"organization": organization, "provider": integration.get_provider()},
+    def post(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        params_dict = kwargs["params"]
+        params = IdentityParams(
+            organization=params_dict["organization"],
+            integration=params_dict["integration"],
+            idp=params_dict["idp"],
+            slack_id=params_dict["slack_id"],
+            channel_id=params_dict["channel_id"],
+        )
```
Comment (ykamo001) [inline src/sentry/integrations/slack/views/link_identity.py:111]: Should we do any key error catches here?

**Review by ykamo001 — APPROVED** [review]
unblocking as you add more changes, functionally looks solid atm!

**Pull request merged**